spec/acceptance/traits_spec.rb

```diff
@@ -293,15 +280,63 @@ def build_user_factory_with_admin_trait(trait_name)
 end
 
 describe "looking up traits that don't exist" do
-  it "raises a KeyError" do
-    define_class("User")
+  context "when passing an invalid override trait" do
+    it "raises a KeyError" do
+      define_class("User")
 
-    FactoryBot.define do
-      factory :user
+      FactoryBot.define do
+        factory :user
+      end
+
+      expect { FactoryBot.build(:user, double("not a trait")) }
+        .to raise_error(KeyError)
+    end
+  end
+
+  context "when the factory includes an invalid default trait" do
+    it "raises a KeyError including the factory name" do
+      define_class("User")
+
+      FactoryBot.define do
+        factory :user do
+          inaccessible_trait
+        end
+
+        factory :some_other_factory do
+          trait :inaccessible_trait
+        end
+      end
+
+      expect { FactoryBot.build(:user) }.to raise_error(
+        KeyError,
+        'Trait not registered: "inaccessible_trait" ' \
+        '(referenced within "user" definition)'
```

Comment:
Was this multi-line string formatted by Standard?

Would a HEREDOC be a viable alternative?

Reply:
Is this what you were thinking?

```rb
        <<~MSG.strip
          Trait not registered: "inaccessible_trait" (referenced within "user" definition)
        MSG
```

At that point I would prefer:

```rb
        'Trait not registered: "inaccessible_trait" (referenced within "user" definition)'
```

I didn't write it that way because the line was getting a bit long. But I could be convinced that it is better for readability. What do you think?

Reply:
I went with the string all on one line for now, since it is a little bit easier to read.